oclc_classify.py

```diff
@@ -0,0 +1,757 @@
+                representation_type, editions = parser.parse(
+                    self._db, swid_xml, **restrictions
+                )
+                if representation_type == parser.SINGLE_WORK_DETAIL_STATUS:
+                    records.extend(editions)
+                elif representation_type in ignored_representation_types:
+                    pass
+                else:
+                    raise IOError(
+                        "Got unexpected representation type from \
+                        lookup: %s" % representation_type
+                    )
+        return records
+
+    def merge_contributors(self, edition, records):
+        """Connect the Gutenberg book to the OCLC works looked up by
+        title/author. Hopefully we can also connect the Gutenberg book
+        to an author who has an LC and VIAF.
+        """
+        # First, find any authors associated with this book that
+        # have not been given VIAF or LC IDs.
+        gutenberg_authors_to_merge = [
+            x for x in edition.author_contributors if not x.viaf or not x.lc
+        ]
+        gutenberg_names = set([x.sort_name for x in edition.author_contributors])
+        for r in records:
+            if gutenberg_authors_to_merge:
+                oclc_names = set([x.sort_name for x in r.author_contributors])
+                if gutenberg_names == oclc_names:
+                    # Perfect overlap. We've found an OCLC record
+                    # for a book written by exactly the same
+                    # people as the Gutenberg book. Merge each
+                    # Gutenberg author into its OCLC equivalent.
+                    for gutenberg_author in gutenberg_authors_to_merge:
+                        oclc_authors = [x for x in r.author_contributors
+                                        if x.sort_name == gutenberg_author.sort_name]
+                        if len(oclc_authors) == 1:
+                            oclc_author = oclc_authors[0]
+                            if oclc_author != gutenberg_author:
+                                gutenberg_author.merge_into(oclc_author)
+                                gutenberg_authors_to_merge.remove(
+                                    gutenberg_author)
+
+            # Now that we've (perhaps) merged authors, calculate the
+            # similarity between the two records.
+            strength = edition.similarity_to(r)
+            if strength > 0:
+                edition.primary_identifier.equivalent_to(
+                    self.output_source, r.primary_identifier, strength
+                )
+
+    def process_item(self, identifier):
+        edition = self.edition(identifier)
+        if isinstance(edition, CoverageFailure):
+            return edition
+
+        # Perform a title/author lookup.
+        title, author, language = self.get_edition_info(edition)
+        if not (title and author):
+            e = 'Cannot lookup edition without title and author!'
```

Comment:
Added a failure for this so we can look up again later if a title/author becomes available.